pkg/config/expand.go

```diff
@@ -504,7 +504,7 @@ type varContext struct {
 	varString  string
 	groupIndex int
 	modIndex   int
-	blueprint  Blueprint
+	dc         DeploymentConfig
```

Comment:
I think (still fuzzy on Go var management) you may need to store a reference to DeploymentConfig here.
It only works now because ModuleConnection is a map and shallow copy of `dc` will point to the same map.

Reply:
Please review and let me know if you believe this is fragile.

Reply:
I think it is, this code relies on tricky "map is shared across copies, because it's a pointer" fact. What if `varContext ` user will attempt to modify other fields in `dc`, or `moduleConnection` will stop being a map.
This concern is applicable for all usages of `varContext` and `applyGlobalVarsInGroup `
See toy example: https://go.dev/play/p/GF3fGiNuvRn

Did I miss something clever that renders my concern void?

Reply:
Nothing clever missed, you just expressed some uncertainty so I didn't prioritize it while fixing other issues. Should be addressed.